base.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```

Comment:
- [ ] Great job so far @byiringiroscar However, based on ruby conventions each class should be in its own file. Kindly move this class to its own file, then import it instead, for example:

`require_relative 'capitalize.rb`